### PR TITLE
Add tests for V8 log and UI classification, fix some Windows bugs and edge cases

### PR DESCRIPTION
**Pull request by AlanSl**
- [x] Add static unit tests, testing strings from real profiles
- [x] Add integration tests generating a profile from a fixture
- [x] Fix issues on Mac / Node 11 (tested on 11.3.0 and 11.0.0) 
- [x] Fix issues on Mac / Node 10 (tested on 10.13.0, 10.12.0 and 10.0.0) 
- [x] Fix issues on Mac / Node 8 (tested on 8.13.0)
- [x] Fix issues on Windows / Node 11 (tested on 11.3.0 and 11.0.0) 
- [x] Fix issues on Windows / Node 10 (tested on 10.13.0, 10.12.0 and 10.0.0)  
- [x] Fix issues on Windows / Node 8 (tested on 8.13.0)
- [x] Check where Node 9 sits regarding version-specific issues
- [x] Update against latest master

This does quite a few things:

 - Brings test coverage up, from almost nothing to >~50% on Node 11 (note that `tap --cov` breaks 0x on Node 8):

```
---------------------------|----------|----------|----------|----------|-------------------|
File                       |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
---------------------------|----------|----------|----------|----------|-------------------|
All files                  |    57.48 |    47.77 |    57.01 |    58.18 |                   |
```

 - Adds unit and integration tests for backend log processing and frame naming
 - Adds unit and integration tests for frontend type classification used by the filter buttons
 - Fixes things (all of which are covered in the tests):
   - Fixes https://github.com/davidmarkclements/0x/issues/195 - app and deps are now correctly recognised on windows
   - Refines the fix to https://github.com/davidmarkclements/0x/issues/192 so that it doesn't only fix Windows paths, it also fixes regex definitions containing \\ on all platforms; and since it is now always applied, cuts out a skippable file write
   - Refines that fix so that it doesn't break certain unicode strings that are written in the initial log file as escaped codes like \\u12a3 if they follow a path separator
   - Fixes a several false-positive edge cases in the frontend type classification (things like, folders that contain ' native ', regexes that contain '.js' or any other identifier)
   - Fixes an issue with backslashes and unicode in regex definitions being over-escaped in Node 8 and appearing garbled in the output

### CODE REVIEW

**Comment by AlanSl** [comment]
Turns out it merges cleanly with latest master, and tests still pass.

Would like @mafintosh to have a look at the changes to the fix for https://github.com/davidmarkclements/0x/issues/192 before merge and, if possible, run the tests on Linux

File: test/v8-types-generated.test.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  }, { actual: 'TemplateLiteral' , expected: 'Literal' })
+
+  t.throws(function () {
+    evalSafeRegexDef("`/abc${ function RegExp () { console.log('>>> Do bad things') } }/`")
+  }, { actual: 'TemplateLiteral' , expected: 'Literal' })
+
+  t.end()
+})
+
+test('Generate profile and test its output', function (outerTest) {
```
[the remark this answers is not in the document]
Comment (AlanSl) [reply test/v8-types-generated.test.js]: I did try that, but I ran into a whole lot of strange issues where Tap was failing intermittently (possibly, depending on how long the test took to run; but the tests were _not_ timing out), with errors caused by it for some reason miscounting the number of expected tests (adding `t.plan(n)` didn't solve it, it just changed the intermittently failure to be Tap miscounting the tests run, instead of miscounting the tests expected). 

I'm not entirely sure why, but adding the outer and inner tests and the .then() chain was about the only thing that reliably prevented these (or at least, prevented them so long as the total test file duration wasn't above around 25 seconds, which should be okay since on an old macbook air they take 10 seconds).

**Comment by AlanSl** [comment]
Added some fixes:

 - Minor thing taking some needless repetition outside the function that is called on every log line
 - Some linting fixes (I'd forgotten that `npm run test` was already set up and called `standard` as well as running the tests)
 - A restructure to use Async/Await

- [x] Test on Node 8, 10 and 11 on Mac
- [ ] Test on Node 8, 10 and 11 on Windows

File: test/v8-types-generated.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlanSl) [reply test/v8-types-generated.test.js]: Thinking about it, these errors were when I had both sets of tests in one file. It was probably just Tap getting confused on timing out while there were still async tests in the pipe.

I've tried it again and it seems pretty reliable now (not tried it on Windows yet...)

**Comment by AlanSl** [comment]
I've tested the updated version on Windows. No problems on Node 11 or Node 10, but had some intermittent timeouts on Node 8, so I've split the slower test file into two.

**Comment by mcollina** [comment]
You can raise tap timeouts from the tap cli.

**Comment by davidmarkclements** [comment]
@AlanSl did we get a confirmation on the timeout issue or is there more to come?

I've added .travis.yml to master - would you mind pulling that into this PR so we can trigger a build for the branch?

**Comment by AlanSl** [comment]
@davidmarkclements No timeouts or tests coming close to the timeout limit after the commit splitting the test files. I've merged and pushed the Travis commits but they're failing with:

```
$ npm run ci
npm ERR! missing script: ci
```

Same error on Master too. Also are you sure we want Travis to run CI on Node 6? Is there a workaround lined up for no async/await support when `zeroEks` is an async function?

**Comment by mcollina** [comment]
Can you please remove node 6, and adjust package.json/travis so that we can have a passing run here?

**Comment by AlanSl** [comment]
@mcollina Done, it now runs `npm run test` on 8, 10, and 8. All green.

**Comment by davidmarkclements** [comment]
awesome nice work @AlanSl
